File: code/jonghyeonjo99/week6/prog_118666.py
def solution(survey, choices):
    answer = ''
    results = [[0 for _ in range(2)] for _ in range(8)]
               
    for i in range(len(choices)):
        if survey[i] == 'RT':
            if 1 <= choices[i] <= 3:
                results[0][0] += 4 - choices[i]
            elif 5 <= choices[i] <= 7:
                results[0][1] += choices[i] - 4
        elif survey[i] == 'TR':
            if 1 <= choices[i] <= 3:
                results[1][0] += 4 - choices[i]
            elif 5 <= choices[i] <= 7:
                results[1][1] += choices[i] - 4
        elif survey[i] == 'FC':
            if 1 <= choices[i] <= 3:
                results[2][0] += 4 - choices[i]
            elif 5 <= choices[i] <= 7:
                results[2][1] += choices[i] - 4
        elif survey[i] == 'CF':
            if 1 <= choices[i] <= 3:
                results[3][0] += 4 - choices[i]
            elif 5 <= choices[i] <= 7:
                results[3][1] += choices[i] - 4
        elif survey[i] == 'MJ':
            if 1 <= choices[i] <= 3:
                results[4][0] += 4 - choices[i]
            elif 5 <= choices[i] <= 7:
                results[4][1] += choices[i] - 4
        elif survey[i] == 'JM':
            if 1 <= choices[i] <= 3:
                results[5][0] += 4 - choices[i]
            elif 5 <= choices[i] <= 7:
                results[5][1] += choices[i] - 4
        elif survey[i] == 'AN':
            if 1 <= choices[i] <= 3:
                results[6][0] += 4 - choices[i]
            elif 5 <= choices[i] <= 7:
                results[6][1] += choices[i] - 4
        elif survey[i] == 'NA':
            if 1 <= choices[i] <= 3:
                results[7][0] += 4 - choices[i]
            elif 5 <= choices[i] <= 7:
                results[7][1] += choices[i] - 4
    
    answer += 'R' if (results[0][0] + results[1][1] >= results[0][1] + results[1][0]) else 'T'
    answer += 'C' if (results[2][0] + results[3][1] <= results[2][1] + results[3][0]) else 'F'
    answer += 'J' if (results[4][0] + results[5][1] <= results[4][1] + results[5][0]) else 'M'
    answer += 'A' if (results[6][0] + results[7][1] >= results[6][1] + results[7][0]) else 'N'
    
    return answer

File: code/SungHo-Ck/week6/prog_118666.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

string solution(vector<string> servey, vector<int> choices) {
    int mbti[2][4] = { {0,0,0,0},{0,0,0,0} };
    char mbti_char[2][4] = { {'R','C','J','A'},{'T','F','M','N'} };
    string answer = "";
    for (int i = 0; i < servey.size(); i++) {
        if (choices[i] < 4) {
            switch (servey[i][0]) {
            case 'R': mbti[0][0] += 4 - choices[i]; break;
            case 'T': mbti[1][0] += 4 - choices[i]; break;
            case 'C': mbti[0][1] += 4 - choices[i]; break;
            case 'F': mbti[1][1] += 4 - choices[i]; break;
            case 'J': mbti[0][2] += 4 - choices[i]; break;
            case 'M': mbti[1][2] += 4 - choices[i]; break;
            case 'A': mbti[0][3] += 4 - choices[i]; break;
            case 'N': mbti[1][3] += 4 - choices[i];
            }
        }
        else if (choices[i] > 4) {
            switch (servey[i][1]) {
            case 'R': mbti[0][0] += choices[i] - 4; break;
            case 'T': mbti[1][0] += choices[i] - 4; break;
            case 'C': mbti[0][1] += choices[i] - 4; break;
            case 'F': mbti[1][1] += choices[i] - 4; break;
            case 'J': mbti[0][2] += choices[i] - 4; break;
            case 'M': mbti[1][2] += choices[i] - 4; break;
            case 'A': mbti[0][3] += choices[i] - 4; break;
            case 'N': mbti[1][3] += choices[i] - 4;
            }
        }
    }
    for (int i = 0; i < 4; i++) {
        if (mbti[0][i] >= mbti[1][i]) {
            answer.append(1, mbti_char[0][i]);
        }
        else {
            answer.append(1, mbti_char[1][i]);
        }
    }
    return answer;
}

int main(void) {
    vector<string> servey = { "AN", "CF", "MJ", "RT", "NA" };
    vector<int> choices = { 5, 3, 2, 7, 5 };
    string ans = solution(servey, choices);
    cout << ans << endl;

    servey = { "TR", "RT", "TR" };
    choices = { 7, 1, 3 };
    ans = solution(servey, choices);
    cout << ans << endl;
    return 0;
}